### PR TITLE
Add short explanation of goal of the rules to README

### PR DESCRIPTION
**Pull request by segiddins**

### CODE REVIEW

**Review by justinseanmartin — APPROVED** [review]
🙇

**Review by thiagohmcruz — APPROVED** [review]

**Review by jerrymarino — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -14,6 +14,13 @@ in Bazel, and by the
 [`swift_library` rule](https://github.com/bazelbuild/rules_swift/blob/master/doc/rules.md#swift_library)
 available from [rules_swift](https://github.com/bazelbuild/rules_swift).
 
+The goal of rules_ios has always been to allow seamlessly transitioning a set of targets from being built
+in Xcode to being built in bazel, with minimal-to-no code changes.
+In that vein, it acts as glue between the underlying rules (rules_apple and rules_swift and the native objc rules),
+applying defaults that mirror what Xcode does. 
+Additionally, rules_ios comes built-in with extension points that act as a quasi-toolchain,
```
Comment (jerrymarino) [inline README.md:21]: Mind expanding on extension points and quasi toolchain features?

File: README.md [hunk 2]
```diff
@@ -14,6 +14,13 @@ in Bazel, and by the
 [`swift_library` rule](https://github.com/bazelbuild/rules_swift/blob/master/doc/rules.md#swift_library)
 available from [rules_swift](https://github.com/bazelbuild/rules_swift).
 
+The goal of rules_ios has always been to allow seamlessly transitioning a set of targets from being built
+in Xcode to being built in bazel, with minimal-to-no code changes.
+In that vein, it acts as glue between the underlying rules (rules_apple and rules_swift and the native objc rules),
```
Comment (jerrymarino) [inline README.md:19]: Also it'd be cool to document was the goal of a "framework packaging" packaging rule and what problems it is solving

File: README.md [hunk 3]
```diff
@@ -14,6 +14,13 @@ in Bazel, and by the
 [`swift_library` rule](https://github.com/bazelbuild/rules_swift/blob/master/doc/rules.md#swift_library)
 available from [rules_swift](https://github.com/bazelbuild/rules_swift).
 
+The goal of rules_ios has always been to allow seamlessly transitioning a set of targets from being built
```
Comment (jerrymarino) [inline README.md:17]: Can you speak to the goals on the Xcode generation side of this repo? There's been a lot interest peaked on it recently and many of the commits here are regarding that

**Review by jerrymarino — COMMENTED** [review]
Adding my

File: README.md (hunk 3 above)
Comment (segiddins) [reply README.md:17]: I'll leave that to amber to expound on

Comment (justinseanmartin) [reply README.md:17]: @gyfelton should be able to help with this too while Amber is out

**Pull request merged**